staging/src/k8s.io/pod-security-admission/policy/registry.go

```diff
@@ -119,28 +139,82 @@ func populate(r *checkRegistry, validChecks []Check) {
 		}
 	}
 
+	var (
+		restrictedChecks = map[api.Version]map[CheckID]VersionedCheck{}
+		baselineChecks   = map[api.Version]map[CheckID]VersionedCheck{}
+
+		baselineIDs, restrictedIDs []CheckID
+	)
 	for _, c := range validChecks {
 		if c.Level == api.LevelRestricted {
-			inflateVersions(c, r.restrictedChecks, r.maxVersion)
+			restrictedIDs = append(restrictedIDs, c.ID)
+			inflateVersions(c, restrictedChecks, r.maxVersion)
 		} else {
-			inflateVersions(c, r.baselineChecks, r.maxVersion)
+			baselineIDs = append(baselineIDs, c.ID)
+			inflateVersions(c, baselineChecks, r.maxVersion)
+		}
+	}
+
+	// Sort the IDs to maintain consistent error messages.
+	sort.Slice(restrictedIDs, func(i, j int) bool { return restrictedIDs[i] < restrictedIDs[j] })
+	sort.Slice(baselineIDs, func(i, j int) bool { return baselineIDs[i] < baselineIDs[j] })
+	orderedIDs := append(baselineIDs, restrictedIDs...) // Baseline checks first, then restricted.
+
+	for v := api.MajorMinorVersion(1, 0); v.Older(nextMinor(r.maxVersion)); v = nextMinor(v) {
+		// Add the filtered baseline checks to restricted.
+		overrides := map[CheckID]bool{}
+		for _, c := range restrictedChecks[v] {
+			for _, override := range c.OverrideCheckIDs {
+				overrides[override] = true
+			}
 		}
+		for id, c := range baselineChecks[v] {
+			if !overrides[id] {
+				restrictedChecks[v][id] = c
```

Comment:
could `restrictedChecks[v]` be nil here if there were baseline checks defined and no restricted checks?

Reply:
Yeah, or if none of the restricted checks had a v1.0 version.

Reply:
I added test cases to cover this.